src/common/cockpitmemory.c

```diff
@@ -113,6 +113,29 @@ strndupx (const char *s,
   return r;
 }
 
+void
+snprintfx (char *str,
```

Comment:
Are you sure about the name of this function?  For me, the `*x` paradigm is for memory-allocating functions that abort on failure to allocate memory.